examples/startup_notification.rs

```diff
@@ -0,0 +1,138 @@
+//! Demonstrates the use of startup notifications on Linux.
+
+fn main() {
+    example::main();
+}
+
+#[cfg(any(x11_platform, wayland_platform))]
+#[path = "./util/fill.rs"]
+mod fill;
+
+#[cfg(any(x11_platform, wayland_platform))]
+mod example {
+    use std::collections::HashMap;
+    use std::rc::Rc;
+    use std::time::{Duration, Instant};
+
+    use winit::event::{ElementState, Event, KeyEvent, StartCause, WindowEvent};
+    use winit::event_loop::EventLoop;
+    use winit::keyboard::Key;
+    use winit::platform::startup_notify::{
+        EventLoopExtStartupNotify, WindowBuilderExtStartupNotify, WindowExtStartupNotify,
+    };
+    use winit::window::{Window, WindowBuilder, WindowId};
+
+    pub(super) fn main() {
+        // Create the event loop and get the activation token.
+        let event_loop = EventLoop::new();
+        let mut current_token = match event_loop.read_token_from_env() {
+            Some(token) => Some(token),
+            None => {
+                println!("No startup notification token found in environment.");
+                None
+            }
+        };
```

Comment:
I think it's good to leave as is just to show what you could do, but we may state that it might not work. I could tune it later myself though.